vowpalwabbit/core/src/reductions/cb/cb_to_cb_adf.cc

```diff
@@ -87,19 +87,12 @@ void predict_or_learn(cb_to_cb_adf& data, multi_learner& base, VW::example& ec)
   else { ec.pred.multiclass = data.adf_data.ecs[0]->pred.a_s[0].action + 1; }
 }
 
-void finish_example(VW::workspace& all, cb_to_cb_adf& c, VW::example& ec)
+void output_example_prediction_cb_to_cb_adf(
+    VW::workspace& all, const cb_to_cb_adf& c, const VW::example& ec, VW::io::logger&)
 {
-  if (c.explore_mode)
-  {
-    c.adf_data.ecs[0]->pred.a_s = std::move(ec.pred.a_s);
-    c.adf_learner->print_example(all, c.adf_data.ecs);
-  }
-  else
-  {
-    c.adf_data.ecs[0]->pred.multiclass = std::move(ec.pred.multiclass);
-    c.adf_learner->print_example(all, c.adf_data.ecs);
-  }
-  VW::finish_example(all, ec);
+  if (c.explore_mode) { c.adf_data.ecs[0]->pred.a_s = std::move(ec.pred.a_s); }
+  else { c.adf_data.ecs[0]->pred.multiclass = std::move(ec.pred.multiclass); }
+  c.adf_learner->print_example(all, c.adf_data.ecs);
```

Comment:
`print_example` as it is today does more than just output prediction. So this output_prediction function should not be calling it.

`cb_to_cb_adf` should implement the 3 (update stats, output pred and print update) and be delegating the calls respectively to the underlying in much the same way it used to do for the the finish->print_example call beforre